sdk/lib/drivers/sound/mmixer/sup.c

```diff
@@ -689,30 +672,25 @@ MMixerSetGetVolumeControlDetails(
     if (!Input)
         return MM_STATUS_UNSUCCESSFUL; /* To prevent dereferencing NULL */
 
+    /* Get maximum available range */
+    MaxRange = VolumeData->SignedMaximum - VolumeData->SignedMinimum;
+
     /* Loop for each channel */
     for (Channel = 0; Channel < MixerControlDetails->cChannels; Channel++)
     {
         if (bSet)
         {
-            /* FIXME SEH */
-            Index = Input[Channel].dwValue / VolumeData->InputSteppingDelta;
-
-            if (Index >= VolumeData->ValuesCount)
-            {
-                DPRINT1("Index %u out of bounds %u \n", Index, VolumeData->ValuesCount);
-                return MM_STATUS_INVALID_PARAMETER;
-            }
-
-            Value = VolumeData->Values[Index];
+            /* Convert from the logical units to the hardware range (DB) */
+            Value = Input[Channel].dwValue * 46.5 - MaxRange + 46;
```

Comment:
Are there other places in this mmixer driver, where floats are being used? Or is it the only place here (and below)?

Reply:
As I see, they exist only in these 2 places (which are introduced by my new changes). I don't remember it existing anywhere else in MMixer code.

Reply:
Using floating point numbers in kernel code isn't usually recommended without taking extra precautions.